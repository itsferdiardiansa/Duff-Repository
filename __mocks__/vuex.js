module.exports = {
  ...jest.requireActual('vuex'),
  __esModule: true,
};
