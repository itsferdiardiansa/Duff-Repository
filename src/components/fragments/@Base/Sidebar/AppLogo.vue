<template>
  <div :class="['app-logo', { mono: isCollapsed }]">
    <router-link to="/" class="app-logo--link">
      <template v-if="!isCollapsed">
        <span> Loket<strong class="font-black">CMS</strong> </span>
      </template>
      <template v-else>
        <span><strong>L</strong></span>
      </template>
    </router-link>
  </div>
</template>
<script>
import { inject } from 'vue';
export default {
  setup() {
    const sidebarContext = inject('sidebarContext');

    return {
      isCollapsed: sidebarContext.isCollapsed,
    };
  },
};
</script>
<style lang="scss" scoped>
.app-logo {
  @apply px-6 py-5 h-16 bg-gray-100;

  &.mono {
    @apply py-4;
    span {
      @apply text-3xl;
    }
  }

  &--link {
    @apply text-xl text-gray-500;
  }
}
</style>
