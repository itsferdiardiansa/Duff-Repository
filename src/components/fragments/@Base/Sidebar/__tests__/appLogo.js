import { mount } from '@vue/test-utils';
import { createStore } from 'vuex';
import { ref } from 'vue';
import AppLogo from '@base/Sidebar/AppLogo';
import appModule from '@store/modules/app';

describe('fragment/Base/Sidebar/AppLogo', () => {
  let store;

  beforeAll(() => {
    store = createStore({
      modules: {
        app: appModule,
      },
    });
  });
  it('will render and match snapshot', async () => {
    const Component = mount(<AppLogo />, {
      global: {
        plugins: [store],
        stubs: ['router-link'],
        provide: {
          isCollapsed: false,
          sidebarContext: {
            isCollapsed: ref(true),
            toggleCollapseSidebar: jest.fn(),
          },
        },
      },
    });

    expect(Component.html()).toMatchSnapshot();
  });

  it('will render collapse view and match snapshot', async () => {
    const Component = mount(<AppLogo />, {
      global: {
        plugins: [store],
        stubs: ['router-link'],
        provide: {
          sidebarContext: {
            isCollapsed: ref(true),
            toggleCollapseSidebar: jest.fn(),
          },
        },
      },
    });

    expect(Component.find('.app-logo.mono').exists()).toBeTruthy();
    expect(Component.html()).toMatchSnapshot();
  });
});
