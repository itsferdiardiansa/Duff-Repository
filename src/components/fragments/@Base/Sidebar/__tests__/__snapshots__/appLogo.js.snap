// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`fragment/Base/Sidebar/AppLogo will render and match snapshot 1`] = `
<div class="app-logo mono">
  <router-link-stub to="/" class="app-logo--link"></router-link-stub>
</div>
`;

exports[`fragment/Base/Sidebar/AppLogo will render collapse view and match snapshot 1`] = `
<div class="app-logo mono">
  <router-link-stub to="/" class="app-logo--link"></router-link-stub>
</div>
`;
